### PR TITLE
GH-323 - Refactor Poam to Task

### PR DESCRIPTION
**Pull request by evdhoudt**
Closes NCI-Agency/anet#323

### CODE REVIEW

File: client/src/components/ReportsByTask.js [hunk 1]
```diff
@@ -44,27 +45,28 @@ export default class ReportsByPoam extends Component {
...
-          xLabel='poam.shortName'
-          onBarClick={this.goToPoam}
+          xLabel='task.shortName'
+          onBarClick={this.goToTask}
           updateChart={this.state.updateChart}
           isLoading={this.state.isLoading}
         />
         <Fieldset
-          title={`Reports by PoAM ${focusDetails.titleSuffix}`}
+          title={`Reports by Task ${focusDetails.titleSuffix}`}
```
Comment (maradragan) [inline client/src/components/ReportsByTask.js]: Here you should also use the taskShortTitle.

File: client/src/components/ReportsByTask.js [hunk 2]
```diff
@@ -79,10 +81,10 @@ export default class ReportsByPoam extends Component {
...
     let resetFnc = ''
     let resetButtonLabel = ''
-    if (this.state.focusedPoam) {
-      titleSuffix = `for ${this.state.focusedPoam.shortName}`
-      resetFnc = 'goToPoam'
-      resetButtonLabel = 'All PoAMs'
+    if (this.state.focusedTask) {
+      titleSuffix = `for ${this.state.focusedTask.shortName}`
+      resetFnc = 'goToTask'
+      resetButtonLabel = 'All tasks'
```
Comment (maradragan) [inline client/src/components/ReportsByTask.js]: And maybe here also use the plural version for taskShortTitle.

File: anet.yml [hunk 3]
```diff
@@ -152,8 +152,12 @@ dictionary:
   ADMINISTRATOR_POSITION_TYPE_TITLE: ANET Administrator
   PRINCIPAL_ORG_NAME: Afghan Government Organization
   ADVISOR_ORG_NAME: Advisor Organization
-  POAM_LONG_NAME: Plan of Action and Milestones / Pillars
-  POAM_SHORT_NAME: PoAM
+  TASK_LONG_NAME: Plan of Action and Milestones / Pillars
+  TASK_SHORT_NAME: Task
```
Comment (maradragan) [inline anet.yml]: I think wou also need the plural for this one, as we sometimes use Tasks (see insights, where we use now All tasks).

File: client/src/components/ReportsByTask.js [hunk 4]
```diff
@@ -106,42 +108,42 @@ export default class ReportsByPoam extends Component {
           }
         }
       `, {chartQueryParams}, '($chartQueryParams: ReportSearchQuery)')
-    const noPoam = {
+    const noTask = {
       id: -1,
-      shortName: 'No PoAM',
-      longName: 'No PoAM'
+      shortName: 'No Task',
+      longName: 'No Task'
```
Comment (maradragan) [inline client/src/components/ReportsByTask.js]: Use here the TASK_SHORT_NAME from the dictionary.

File: src/main/resources/emails/rollup.ftl [hunk 5]
```diff
@@ -153,13 +153,13 @@ a {
...
-                <strong>PoAM:</strong> ${(poam.shortName)!} ${(poam.longName)!}
+            <#-- <a href="${serverUrl}/tasks/${task.id}"> -->
+<!--                <img height="20" alt="Task" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABkAAAAXCAYAAAD+4+QTAAAABmJLR0QA/wDWABipT60nAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAB3RJTUUH4QERBQskHTWEPAAAABl0RVh0Q29tbWVudABDcmVhdGVkIHdpdGggR0lNUFeBDhcAAAGnSURBVEjHzZUxqsJAEIZnAtlgFYM22huJd9AriHYK2uoBFG+QRi+TSgtNoZUi2wUxsRMU21zgf5XyghoTUd5bGLbYnf3499+ZZQCgLw8lbcJ2u01PQYoxnU5RrVYxm83SpCGVEtd1abVakeu631GyXq9RKBRARCgWi9hsNp9X4jgOnc9nYmY6nU7kOM5nlRyPR+i6DiK6RTabxeVySaSEHz3h5XJJvu/TbrejIAhov9+T7/vEzATgNpumSeVymUqlElmWRaZpUq1Wi1di2zY0TUMmk4GmaRBCQFVVMHNEBRGBmcHMUFUVQohInm3bESV319Xr9aAoysOD44KZoSgK+v3+3XU99KTT6UAIkRjEzBBCoNvtPvTkqfGtVisR6Apot9tPjY99XfV6PRGk0Wi8XyfNZjNRGbzaFwuRUhIzv4RIKd+HeJ73EsLM5Hne+5DD4UC/a7VSqdBoNCLLsiL7giB4v61cW4lhGJhMJpG18XgMwzBARNB1Pdb4p5D5fI5cLofhcBh7wGAwQD6fx2KxSA+RUiIMw0QNMAxDSCnTNcg//+P/LeQHMm6FWr8JMB8AAAAASUVORK5CYII=" /> -->
+                <strong>Task:</strong> ${(task.shortName)!} ${(task.longName)!}
```
Comment (maradragan) [inline src/main/resources/emails/rollup.ftl]: Here we should actually also use the TASK_SHORT_NAME from the dictionary.

File: src/main/resources/migrations.xml [hunk 6]
```diff
@@ -812,4 +812,68 @@
 		</addColumn>
 	</changeSet>
 
+	<changeSet id="renamePoams-to-Tasks" author="evdhoudt">
+		<!-- Tasks -->
+		<renameTable
+				newTableName="tasks"
+				oldTableName="poams" />
```
Comment (maradragan) [inline src/main/resources/migrations.xml]: indentation is not ok.

File: anet.yml (hunk 3 above)
Comment (VassilIordanov) [reply anet.yml]: Maybe for now we can get away by doing a + "s" when plural is needed

Comment (evdhoudt) [reply anet.yml]: For now I agree with Vasil, maybe at some point a pluralize method needs to be added...

File: client/src/components/ReportsByTask.js (hunk 2 above)
Comment (evdhoudt) [reply client/src/components/ReportsByTask.js]: resetFnc is a callback function, thus should not be linked to the dictionary

File: anet.yml [hunk 7]
```diff
@@ -152,8 +152,12 @@ dictionary:
...
+  TASK_PROJECTED_COMPLETION: 'Fill in the projected completion date'
```
Comment (VassilIordanov) [inline anet.yml]: It seems to me that this shoube called TASK_PROJECTED_COMPLETION_PLACEHOLDER, and there should also be a TASK_PROJECTED_COMPLETION_LABEL, in accordance to whatever format is adopted in GH-311

File: anet.yml [hunk 8]
```diff
@@ -152,8 +152,12 @@ dictionary:
...
+  TASK_PLANNED_COMPLETION: 'Fill in the planned completion date'
+  TASK_CUSTOM_ENUM_LABEL: 'Project status'
+  TASK_CUSTOM_FIELD: 'Custom field'
```
Comment (VassilIordanov) [inline anet.yml]: There should be a TASK_CUSTOM_FIELD_PLACEHOLDER

File: anet.yml (hunk 3 above)
Comment (VassilIordanov) [reply anet.yml]: @maradragan - to lower the threshold for rename detection, in this case you can do something like `git diff -M40% candidate..GH-323`. Unfortunately, I don't know how to tell github to do this, so command line only I guess

**Pull request merged**